src/Migrations/M033MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,286 @@
+                    }
+
+                    // Transform the Social Media block into an Embed block.
+                    $block = self::transform_blocks($block, $social_media_url);
+                }
+
+                // Unset the reference to prevent potential issues.
+                unset($block);
+
+                // Serialize the blocks content.
+                $new_content = serialize_blocks($blocks);
+
+                // We don't update the block if the new content is the same as before.
+                if ($post->post_content === $new_content) {
+                    continue;
+                }
+
+                $post_update = array(
+                    'ID' => $current_post_id,
+                    'post_content' => $new_content,
+                );
+
+                // Update the post with the replaced blocks.
+                $result = wp_update_post($post_update);
+
+                if ($result === 0) {
+                    throw new \Exception("There was an error trying to update the post #" . $current_post_id);
+                }
+
+                echo "Migration successful\n";
+            }
+        } catch (\Exception $e) {
+            // Catch any exceptions and display the post ID if available
+            echo "Migration wasn't executed for post ID: ", $current_post_id ?? 'unknown', "\n";
+            echo $e->getMessage(), "\n";
+        }
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Get the source of a Social Media block.
+     * It can be Instagram, Facebook, or Twitter/X.
+     *
+     * @param string $url - The social media URL used to identify the source.
+     * @return string - The source of the Social Media block or an empty string if no source was found.
+     */
+    private static function identify_source(string $url): string
+    {
+        foreach (self::SOURCE_PATTERNS as $value => $key) {
+            if (preg_match($key, $url)) {
+                return $value;
+            }
+        }
+        return '';
+    }
+
+    /**
+     * Transform a Social Media block into a group of blocks.
+     * This group contains a header for the Social Media block title,
+     * a paragraph for the Social Media block description,
+     * and a final block (Embed) for the Twitter/X embed.
+     *
+     * @param array $block - A parsed Social Media block.
+     * @param string $source - The source of the Social Media block (Twitter or X).
+     * @param string $social_media_url - The Social Media URL.
+     * @return array - The transformed block.
+     */
+    private static function transform_blocks(array $block, string $social_media_url): array
+    {
+        $block_title = array_key_exists("title", $block['attrs']) ? $block['attrs']['title'] : null;
+        $block_description = array_key_exists("description", $block['attrs']) ? $block['attrs']['description'] : null;
+
+        $block = [];
+        $block['blockName'] = Utils\Constants::BLOCK_GROUP;
```

Comment:
Can you please add a function to the `Utils/Functions.php` file to make the creation of Group blocks reusable?